src/components/MultiHopTrade/components/TradeInput/components/Claim/hooks/useArbitrumClaimTx.tsx

```diff
@@ -0,0 +1,131 @@
+import { ethChainId } from '@shapeshiftoss/caip'
+import { CONTRACT_INTERACTION } from '@shapeshiftoss/chain-adapters'
+import { KnownChainIds } from '@shapeshiftoss/types'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import { useMutation, useQuery } from '@tanstack/react-query'
+import { outboxAbi } from 'contracts/abis/Outbox'
+import type { Hash } from 'viem'
+import { encodeFunctionData, getAddress } from 'viem'
+import { queryClient } from 'context/QueryClientProvider/queryClient'
+import { useEvmFees } from 'hooks/queries/useEvmFees'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { getEthersV5Provider } from 'lib/ethersProviderSingleton'
+import {
+  assertGetEvmChainAdapter,
+  buildAndBroadcast,
+  createBuildCustomTxInput,
+} from 'lib/utils/evm'
+import { assertGetViemClient } from 'lib/viem-client'
+
+import type { ClaimDetails } from './useArbitrumClaimsByStatus'
+
+export const useArbitrumClaimTx = (
+  claim: ClaimDetails,
+  setClaimTxHash: (txHash: string) => void,
+  setClaimTxStatus: (txStatus: TxStatus) => void,
+) => {
+  const wallet = useWallet().state.wallet
+
+  const l2Provider = getEthersV5Provider(KnownChainIds.ArbitrumMainnet)
+
+  const executeTransactionDataResult = useQuery({
+    queryKey: ['executeTransactionData', { txid: claim.tx.txid }],
+    queryFn: async () => {
+      const { event, message } = claim
+
+      const proof = await message.getOutboxProof(l2Provider)
+
+      if (!('position' in event)) return
+
+      // nitro transaction
```

Comment:
classic transaction was quite a bit more complicated. thinking we just support nitro unless there is an important need for classic transactions (before Aug-31-2022 )

Reply:
@shapeshift/product

Reply:
That sounds sane to me if the goal is to make the swapper work - having users with legacy classic Txs made outside of ShapeShift that need claim indeed seems like an edge-case to me.